packages/deploy-script-support/src/extract-proposal.js

```diff
@@ -43,16 +46,22 @@ const findModule = (initDir, srcSpec) =>
 /**
  * @param {{ bundleID?: string, bundleName?: string }} handle - mutated then hardened
  * @param {string} sourceSpec - the specifier of a module to load
- * @param {number} sequence - the sequence number of the proposal
+ * @param {PropertyKey} key - the key of the proposal
  * @param {string} piece - the piece of the proposal
  * @returns {Promise<[string, any]>}
  */
-const namedHandleToBundleSpec = async (handle, sourceSpec, sequence, piece) => {
-  handle.bundleName = `coreProposal${sequence}_${piece}`;
+const namedHandleToBundleSpec = async (handle, sourceSpec, key, piece) => {
+  handle.bundleName = `coreProposal${String(key)}_${piece}`;
   harden(handle);
   return harden([handle.bundleName, { sourceSpec }]);
 };
 
+/**
+ * @param {unknown[] | Record<PropertyKey, unknown>} obj
+ */
+const getKeys = obj =>
+  Reflect.ownKeys(obj).filter(key => key !== 'length' || !Array.isArray(obj));
```

Comment:
Do we actually want to support the esoteric array cases? If we don't care, maybe?

```suggestion
  Array.isArray(obj) ? [...Array.prototype.keys.call(obj)] : Reflect.ownKeys(obj);
```